src/helpers/figures/charts/chart_ui_common.ts

```diff
@@ -39,27 +44,33 @@ export function chartToImage(
   canvas.setAttribute("height", figure.height.toString());
   // we have to add the canvas to the DOM otherwise it won't be rendered
   document.body.append(div);
+  let imgContent: string | undefined = undefined;
+  let extensionsLoaded = false;
   if ("chartJsConfig" in runtime) {
+    extensionsLoaded = areChartJSExtensionsLoaded();
+    if (!extensionsLoaded) {
+      registerChartJSExtensions();
+    }
     const config = deepCopy(runtime.chartJsConfig);
     config.plugins = [backgroundColorChartJSPlugin];
     const chart = new window.Chart(canvas, config as ChartConfiguration);
-    const imgContent = chart.toBase64Image() as string;
+    imgContent = chart.toBase64Image() as string;
     chart.destroy();
     div.remove();
-    return imgContent;
   } else if (type === "scorecard") {
     const design = getScorecardConfiguration(figure, runtime as ScorecardChartRuntime);
     drawScoreChart(design, canvas);
-    const imgContent = canvas.toDataURL();
+    imgContent = canvas.toDataURL();
     div.remove();
-    return imgContent;
   } else if (type === "gauge") {
     drawGaugeChart(canvas, runtime as GaugeChartRuntime);
-    const imgContent = canvas.toDataURL();
+    imgContent = canvas.toDataURL();
     div.remove();
-    return imgContent;
   }
-  return undefined;
+  if (!extensionsLoaded) {
+    unregisterChartJsExtensions();
+  }
```

Comment:
this seems wrong no ? this will unload the chartJExtensions if they were loaded before calling this helper, and we only converted gauge/scorecards. The load/unload should probably be fully inside the `if("chartJsConfig" in runtime)`

Reply:
true